source/common/memory/utils.cc

```diff
@@ -18,22 +21,45 @@ void Utils::releaseFreeMemory() {
 
 /*
   The purpose of this function is to release the cache introduced by tcmalloc,
-  mainly in xDS config updates, admin handler, and so on. all work on the main thread,
+  mainly in xDS config updates, admin handler, and so on. All work on the main thread,
   so the overall impact on performance is small.
   Ref: https://github.com/envoyproxy/envoy/pull/9471#discussion_r363825985
 */
 void Utils::tryShrinkHeap() {
 #ifdef TCMALLOC
-  // TODO(zyfjeff): Make max unfreed memory byte configurable
-  static const uint64_t MAX_UNFREED_MEMORY_BYTE = 100 * 1024 * 1024;
+  uint64_t shrink_threshold = 100 * 1024 * 1024UL;
   auto total_physical_bytes = Stats::totalPhysicalBytes();
   auto allocated_size_by_app = Stats::totalCurrentlyAllocated();
+  Runtime::Loader* runtime = Runtime::LoaderSingleton::getExisting();
+  if (runtime) {
+    shrink_threshold = runtime->threadsafeSnapshot()->getInteger(
+        "envoy.memory.heap_shrink_threshold", shrink_threshold);
+    Envoy::Stats::Scope& root_scope = runtime->getRootScope();
+
+    Envoy::Stats::StatNameManagedStorage shrink_attempt_stat_name("envoy.memory_shrink_attempt",
+                                                                  root_scope.symbolTable());
+
+    Envoy::Stats::Counter& shrink_attempt_counter =
+        root_scope.counterFromStatName(shrink_attempt_stat_name.statName());
```

Comment:
@jmarantz Is this stat name look up a noticeable overhead? How can I save it? 

If it's not a concern I will just fix the `/stats/recentlookups` test to ignore this look up